source/peer-handshake.js

```diff
@@ -47,6 +47,21 @@ Skylink.prototype._doOffer = function(targetMid, iceRestart) {
     }
   }
 
+
+  /* Safari does not support createOffer with (offerToReceiveAudio: 1)
+   * So the following patch is needed until it is fixed in AdapterJS.
+   * AdapterJS ticket : https://github.com/webrtc/adapter/issues/661
+   */
+  if(window.webrtcDetectedBrowser === 'safari' && webrtcDetectedVersion >= 11 
+      && AdapterJS.webrtcDetectedType !== 'plugin'){
+      if (offerConstraints.offerToReceiveAudio) {
```

Comment:
So I understand that this is based off [the issue comment solution](https://github.com/webrtc/adapter/issues/661#issuecomment-328518588) for it to work, that however, I have a few concerns:

1. Should I add a transceiver if I already have an existing audio or video track added to the PC since webrtc-adapter polyfill does `addTrack()` which adds a RTCRtpTransceiver?
2. Should I add transceivers again when I'm doing a re-negotiation of offer / answer? Wouldn't it result in additional m= lines?